Mostly ride request related

Omarsh1 opened the pull request:
**Here's what happened**

- Added require lines for the ride js files in application.js
- Bootstrapped the container for request ride page
- Added route in routes.rb for request ride
- Added directory “ride” for all ride related pages and moved request_ride.html.erb there
- Added ride controller
- If logged in as rider redirects to request a ride 
- Before action in ride controller will not allow any other user type (not rider) to use that page, also no not-signed in users are allowed to that page
- Added top nav bar on request ride page
- Added functionality to Logout nav bar item on request a ride page
- Re-direction to account info when user presses Account nav bar item
- Added more steps to the progress bar on request a ride page (Passengers, Drop Off Locations etc.)

xaviablaza commented:
The routes work, but there is no initialized controller for `ride`. To fix this, I did a `docker exec -it csafeapp_app_1 rails g controller Ride`.

xaviablaza commented:
Nice job on the page! I've added a fix for the following and everything should now work as intended.

- Before action in ride controller will not allow any other user type (not rider) to use that page, also no not-signed in users are allowed to that page

Merge when you are ready.

khafa100 commented on csafe-app/app/views/ride/request_ride.html.erb:
```diff
@@ -141,7 +163,76 @@
+          }
+      }
+  </script>
+
```
I know this is a bit nit-picky, but for the sake of best practices and easy debugging in the future, all css and javascript should go into their own designated files and not on live on an html page. Especially since these deal with the navigation menu and will be used throughout the site on different pages. Can you please move these? I'll give a better review when I get home tonight but I just wanted to make sure to mention this before anything is merged.

xaviablaza replied:
To do this, move your javascript to `map_api.js` or `ride.coffee`. Then, copy the CSS into `ride.scss`.

Omarsh1 replied:
It's a personal view on best practice. I think it's easier to debug a page when all related css and javascript live in one folder. But I'll make the changes!

Omarsh1 commented:
Just added rideController and ride related missing files. Please re-approve or som

xaviablaza commented:
Continues to work as intended. The commit below is an alternative to checking the auth for your reference:

https://github.com/andreperkins12/CSAFE-Chapman-University/pull/6/commits/0dc24f88543b5f72a1d166a125f88ca8599dd1bd

Omarsh1 commented:
**Here's what happened since comments on first request**
- Re-structured code so that js, css and html are separate
- Added //= require_tree . to appliaction.js so we don’t have to require each js file we add
- Added more field sets that hide/show when “Request” is pressed on requestRide page
- Made progress bar functional to show current progress of requesting a ride

xaviablaza commented:
Good to go after code reorganization.
